### PR TITLE
Prepare open sourcing

### PR DESCRIPTION
**Pull request by abailly-iohk**
This PR adds some minimal instructions on how to build and use the library, as well as a couple of files relevant to OSS projects: License, contributing guidelines, changelog, and code of conduct.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Contributing to Hydra
```
Comment (ch1bo) [inline CONTRIBUTING.md:1]: Oops 😬

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+When contributing code, it helps to have discussed the rationale and (ideally)
+how something is implemented in a feature idea or bug ticket beforehand.
+
+### Building & Testing
+
+See the [README.md](./README.md#building) file for instructions on how to build and test this package.
+
+### Coding standards
+
+TBD
```
Comment (ch1bo) [inline CONTRIBUTING.md:54]: I think it's better to remove any not yet defined sections.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to  [Package Versioning Policy](https://pvp.haskell.org/) (See this [FAQ](https://pvp.haskell.org/faq/#semver) for a comparison between PVP and [SemVer](https://semver.org/).
```
Comment (ch1bo) [inline CHANGELOG.md:6]: Why PVP?

File: quickcheck-dynamic.cabal [hunk 4]
```diff
@@ -1,63 +1,80 @@
...
+license:         Apache-2.0
 license-files:
   LICENSE
   NOTICE
-maintainer: ulf.norell@quviq.com
-author: Ulf Norell
-homepage: https://github.com/iohk/plutus#readme
-bug-reports: https://github.com/iohk/plutus/issues
+
+maintainer:      ulf.norell@quviq.com
```
Comment (ch1bo) [inline quickcheck-dynamic.cabal:9]: This should maybe be IOHK now.

**Review by ch1bo — CHANGES_REQUESTED** [review]
Remove some hints where you got the templates from 🙂

File: CONTRIBUTING.md (hunk 1 above)
Comment (abailly-iohk) [reply CONTRIBUTING.md:1]: 🤦

File: quickcheck-dynamic.cabal (hunk 4 above)
Comment (abailly-iohk) [reply quickcheck-dynamic.cabal:9]: I have asked  Ulf and Maximilian to join the project in order to get their approval.

**Comment by abailly-iohk** [comment]
@ch1bo Can you review this again? It's more stuff than initially planned and I think it's fine for merging.

**Review by ch1bo — APPROVED** [review]

**Pull request merged**